File: cpp_e9zMZ/cpp/05-uocso-boiso/108.cpp
#include <iostream>
using namespace std;

int main() {
    int number = 3;
    int result = 1;

    int index = number - 1;
    while (index <= number) {
        if (number % index == 0 && index % 2 == 1) {
            result = index;
            break;
        }
        index--;
    }

    cout << "Ước lẻ lớn nhất của " << number << " là " << result;
    return 0;
}

File: python_ToSXs/python/05-uocsoboiso/108.py
number = 27

result = 1
start = number - 1
if start % 2 == 0:
    start -= 1

for i in range(start, 1, -2):
    if number % i == 0:
        result = i
        break

print(f"Ước số lẻ lớn nhất của {number} là {result}")
